src/Psl/DateTime/Timestamp.php

```diff
@@ -0,0 +1,305 @@
+     * @param null|FormatDateStyle $date_style Specifies the style of the date portion for parsing. Defaults to {@see DateFormatStyle::default()}.
+     * @param null|FormatTimeStyle $time_style Specifies the style of the time portion for parsing. Defaults to {@see TimeFormatStyle::default()}.
+     * @param null|FormatPattern|string $pattern Specifies a custom format pattern to be used for parsing. A default, implementation-specific pattern is used if null.
+     * @param null|Timezone $timezone Determines the timezone for which the date and time are parsed. Defaults to {@see Timezone::default()}.
+     * @param null|Locale $locale Specifies the locale to be used for parsing locale-specific date and time components. Defaults to {@see Locale::default()}.
+     *
+     * @throws Exception\RuntimeException If failed to parse the given raw string.
+     *
+     * @return static Returns an instance of {@see Timestamp} representing the parsed date and time.
+     *
+     * @mutation-free
+     *
+     *@see https://unicode-org.github.io/icu/userguide/format_parse/datetime/#datetime-format-syntax
+     */
+    public static function parse(
+        string $raw_string,
+        null|FormatDateStyle $date_style = null,
+        null|FormatTimeStyle $time_style = null,
+        null|FormatPattern|string $pattern = null,
+        null|Timezone $timezone = null,
+        null|Locale $locale = null,
+    ): static {
+        $formatter = Internal\create_intl_date_formatter($date_style, $time_style, $pattern, $timezone, $locale);
+
+        $timestamp = $formatter->parse($raw_string);
+        if ($timestamp === false) {
+            // Only show pattern in the exception if it was provided.
+            // TODO(azjezz): should we have `ParserException` which contains all passed parameters ( styles, locale, timezone, pattern .. etc)?
+            if (null !== $pattern) {
+                $formatter_pattern = $pattern instanceof FormatPattern ? $pattern->value : $pattern;
+
+                throw new Exception\RuntimeException(Str\format(
+                    'Parsing error: Unable to interpret \'%s\' as a valid date/time using pattern \'%s\'.',
+                    $raw_string,
+                    $formatter_pattern,
+                ));
+            }
+
+            throw new Exception\RuntimeException(
+                "Parsing error: Unable to interpret '$raw_string' as a valid date/time.",
+            );
+        }
+
+        /** @psalm-suppress MissingThrowsDocblock */
+        return self::fromRaw((int) $timestamp);
+    }
+
+    /**
+     * Returns this Timestamp instance itself, as it already represents a timestamp.
+     *
+     * @mutation-free
+     */
+    public function getTimestamp(): self
+    {
+        return $this;
+    }
+
+    /**
+     * Returns the raw seconds and nanoseconds of the timestamp as an array.
+     *
+     * @return array{int, int<0, 999999999>}
+     *
+     * @mutation-free
+     */
+    public function toRaw(): array
```

Comment:
is 'raw' the best name we could imagine here?

Reply:
`toParts()` ?

Reply:
that would be more consistent with DateTime indeed